### DISCUSSION TITLE
GNNExplainer for Graph Level Prediction Task

### DISCUSSION THREAD

**Discussion by amoschoomy** [pyg-team/pytorch_geometric#9478] [Q&A] [answered]
```   # Initialize the Explainer
    explainer = Explainer(
        model=model,
        algorithm=GNNExplainer(epochs=200),
        explanation_type='model',
        node_mask_type='attributes',
        model_config=dict(
            mode='binary_classification',
            task_level='graph',
            return_type='probs',
        ))

    sample_reprensatative_pred = []
    sample_reprensatative_label = []
    for i, sample in enumerate(loader):  # iterate over samples
        sample_scores = []
        sample_labels = []
        for data in sample:  # iterate over TCR sequences in each sample
            with torch.no_grad():
                out = model(data) 
                probabilities = torch.sigmoid(out)
                sample_scores.extend(probabilities[:, 1].tolist())
                sample_labels.extend(data.y.tolist())
```

How would one use an explainer to explain my graph level prediction task?

**Comment by rusty1s** [comment, marked as answer]
You can call the explainer similar to how you call the model, i.e.:
```
for i, sample in enumerate(loader):  # iterate over samples
        for data in sample:  # iterate over TCR sequences in each sample
                explanation = explainer(data.x, data.edge_index, ...)
```
Two things:
* The `explainer` can not operate on `data` objects. You need to change your model `forward` function to expect `Tensor` attributes instead: `def forward(self, data)` -> `def forward(self, x, edge_index, ...)`
* `return_type` should be `log_probs` in your case since the model does not seem to output probabilities (you have a `sigmoid` applied afterwards).
